cotainr/tests/conftest.py

```diff
@@ -92,18 +92,25 @@ def data_log_level_names_mapping():
 @pytest.fixture
 def factory_mock_input():
     """
-    Create mock of the builtins `input` function that returns a fixed "input".
+    Create mock of the builtins `input` function that returns an "input".
 
     Returns a factory for creating mocked versions of the builtin `input`
     function to be used with the `monkeypatch` fixture to replace
     `builtins.input` with a function that prints the prompt (its argument, if
-    provided) and returns a "fixed user input", provided as argument to the
+    provided) and returns the next "user input", provided as argument to the
     factory.
     """
 
-    def create_mock_input(fixed_user_input=None):
+    def create_mock_input(fixed_user_input=""):
+        if isinstance(fixed_user_input, list):
+            inputs = iter(fixed_user_input)
+        else:
+            inputs = iter([fixed_user_input])
+
         def mock_input(prompt):
             print(prompt, end="")
+            if isinstance(fixed_user_input, list):
+                return next(inputs)
             return fixed_user_input
```

Comment:
I think this implementation is pretty confusing:
* if `fixed_user_input` is a list, assign `inputs` to an iterator over it and use it
* otherwise, assign `inputs` to an iterator over a single instance of the string, *but never use it* since the `if isinstance(fixed_user_input, list):` within `mock_input` is always taken?

Reply:
Ah i overlooked the if-statement. I'll remove it and just return next(inputs)